cocos/base/CCDirector.h

```diff
@@ -488,6 +489,12 @@ class CC_DLL Director : public Ref
      */
     void resetMatrixStack();
 
+    /**
+     * returns the cocos2d thread id.
+     Useful to know if certain code is already running on the cocos2d thread
+     */
+    const std::thread::id& getCocos2dThreadId() const { return _cocos2d_thread_id; }
```

Comment:
@ricardo
Sorry, the unsquashed PR doesn't pass through jenkins Because lua/js binding can not bind `std::thread::id`. I mistook to check the result of other pull request.

 If this method needs to be exported to script, then should bind this function manually. If this function is not needed in script, then we can just ignore it.